tests/report_tests/plugin_tests/logs.py

```diff
@@ -51,22 +51,26 @@ def pre_sos_setup(self):
         # write over 100MB to ensure we will actually size limit inside sos,
         # allowing for any compression or de-dupe systemd does
         from systemd import journal
-        rsize = 20 * 1048576
-        for i in range(3):
-            # generate 20MB, write it, then write it in reverse.
+        sosfd = journal.stream('sos-testing')
+        rsize = 10 * 1048576
+        for i in range(6):
+            # generate 10MB, write it, then write it in reverse.
             # Spend less time generating new strings
             rand = ''.join(random.choice(ascii_uppercase + digits) for _ in range(rsize))
-            journal.send(rand)
-            journal.send(rand[::-1])
+            sosfd.write(rand + '\n')
+            # sleep to avoid burst rate-limiting
+            sleep(15)
```

Comment:
Was this sleep actually required?

Reply:
Not strictly, but that's more due to how long it takes to generate the test data strings. Previously, we were hitting the burst limit rates when the 20MB strings got split up into X number of messages, the change to stream alleviated that somewhat but not completely. We'd hit the limits somewhere in the second write (when we reverse the generated string). If we had prepared strings that were just rapid-fired off, we'd hit those burst limits again.

So at the moment, we could remove it, but if some change happens that increases the performance of generating that random test data, we could hit burst limits again.

We could also reduce the sleep timer here though if you'd prefer. I chose 15 seconds here based on the journald default of 30 seconds for the burst window and the previous behavior seen.

Reply:
No strong preferences